src/mlpack/methods/ann/loss_functions/earth_mover_distance.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file earth_mover_distance.hpp
+ * @author Shikhar Jaiswal
+ *
+ * Definition of the earth mover distance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTIONS_EARTH_MOVER_DISTANCE_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTIONS_EARTH_MOVER_DISTANCE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The earth mover distance function measures the network's performance
+ * according to the Kantorovich-Rubinstein duality approximation.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class EarthMoverDistance
```

Comment:
I know this is a super simple loss function, but do you mind to add a test to: https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/loss_functions_test.cpp.